### PR TITLE
drivers/cpuid: fixed and unified CPUID driver implementations

### PR DESCRIPTION
**Pull request by haukepetersen**
Besides some minor cleanup, this introduces two major things:
- renamed the `CPUID_ID_LEN` to `CPUID_LEN`, which should be slightly easier to read and just looks nicer :-)
- moved the per CPU definition of `CPUID_LEN` from `cpu_conf.h` to `periph_cpu.h`. This is more consistent, as the CPU specific properties regarding vendor peripherals are all in this file.

### CODE REVIEW

File: cpu/nrf51/include/periph_cpu.h [hunk 1]
```diff
@@ -26,6 +26,18 @@ extern "C" {
 #endif
 
 /**
+ * @brief   Mandatory macro for defining GPIO pins
+ *
+ * The port definition is used (and zeroed) to suppress compiler warnings
+ */
+#define GPIO_PIN(x,y)       ((x & 0) | y)
```
Comment (authmillenon) [inline cpu/nrf51/include/periph_cpu.h]: Unrelated change?

**Comment by authmillenon** [comment]
Nitpicky remark: some of the `CPUID_LEN` values are unsigned (e.g. `8U`) while others are signed (e.g. `8`).

File: cpu/nrf51/include/periph_cpu.h (hunk 1 above)
Comment (haukepetersen) [inline cpu/nrf51/include/periph_cpu.h]: There might be some minor unrelated (though the commit states that some cleanup is included...), but it would take me days to make every single fix into it's own commit/PR...

**Comment by haukepetersen** [comment]
Good catch, I will adjust the non unsigned ones...

File: cpu/kinetis_common/include/periph_cpu.h [hunk 2]
```diff
@@ -54,6 +54,11 @@ typedef uint16_t gpio_t;
 #define GPIO_PIN(port, pin)          ((port << GPIO_PORT_SHIFT) | pin)
 
 /**
+ * @brief   Length of the CPU_ID in octets
+ */
+#define CPUID_LEN           (16)
```
Comment (authmillenon) [inline cpu/kinetis_common/include/periph_cpu.h]: 16U

File: cpu/lm4f120/include/periph_cpu.h [hunk 3]
```diff
@@ -25,7 +25,11 @@
 extern "C" {
 #endif
 
-/* nothing to do here, yet */
+/**
+ * @brief   Length of the CPU_ID in octets
+ */
+#define CPUID_LEN                       (12)
```
Comment (authmillenon) [inline cpu/lm4f120/include/periph_cpu.h]: 12U

File: cpu/native/include/periph_cpu.h [hunk 4]
```diff
@@ -25,7 +25,12 @@
 extern "C" {
 #endif
 
-/* nothing defined here so far... */
+/**
+ * @brief   Length of the CPU_ID in octets
+ */
+#ifndef CPUID_LEN
+#define CPUID_LEN           (4)
```
Comment (authmillenon) [inline cpu/native/include/periph_cpu.h]: 4U

File: cpu/nrf51/include/periph_cpu.h [hunk 5]
```diff
@@ -26,6 +26,18 @@ extern "C" {
...
+
+/**
+ * @brief   Length of the CPU_ID in octets
+ */
+#define CPUID_LEN           (8)
```
Comment (authmillenon) [inline cpu/nrf51/include/periph_cpu.h]: 8U

File: cpu/nrf52/include/periph_cpu.h [hunk 6]
```diff
@@ -56,6 +56,11 @@ typedef enum {
  */
 #define GPIO_PIN(x,y)       ((x & 0) | y)
 
+/**
+ * @brief   Length of the CPU_ID in octets
+ */
+#define CPUID_LEN           (8)
```
Comment (authmillenon) [inline cpu/nrf52/include/periph_cpu.h]: 8U

File: cpu/sam3/include/periph_cpu.h [hunk 7]
```diff
@@ -56,6 +56,11 @@ typedef uint32_t gpio_t;
 /** @} */
 
 /**
+ * @brief   Length of the CPU_ID in octets
+ */
+#define CPUID_LEN           (16)
```
Comment (authmillenon) [inline cpu/sam3/include/periph_cpu.h]: etc.

**Comment by authmillenon** [comment]
> Good catch, I will adjust the non unsigned ones...

Oops, sorry. Thought you already did ^^"

**Comment by haukepetersen** [comment]
Now they should be all unsigned.

**Comment by authmillenon** [comment]
ACK, please squash

**Comment by haukepetersen** [comment]
squashed

**Comment by haukepetersen** [comment]
Travis showed two minor issues, fixed these and squashed again.

**Comment by authmillenon** [comment]
Changes still look fine to me: Re-ACK

**Comment by haukepetersen** [comment]
Travis is happy -> and go
